src/Symfony/Component/Form/Extension/Core/Type/TimezoneType.php

```diff
@@ -27,6 +30,41 @@ class TimezoneType extends AbstractType implements ChoiceLoaderInterface
      * @var ArrayChoiceList
      */
     private $choiceList;
+    private $regions;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $this->regions = $options['regions'];
+
+        if ('datetimezone' === $options['input']) {
+            $builder->addModelTransformer(new CallbackTransformer(function ($timezone) {
```

Comment:
I think to create an individual class for this model transformer it's the common implementation (i.e. into core) and both will be easier to maintain. `CallbackTransformer` it's commonly used in tests to simplify.